src/backend/index.js

```diff
@@ -68,26 +67,27 @@ status.startHeartbeat();
  * card. It is a shared folder that the user can access. The data folder
  * accessible through rnBridge.app.datadir() is not accessible by the user.
  *
+ * Other config data such as the path to the APK file is only available in the
+ * React Native process, so we pass that to the backend here
+ *
  * We need to wait for the React Native process to tell us where the folder is.
- * This code supports re-starting the server with a different folder if
- * necessary (we probably shouldn't do that)
  */
-rnBridge.channel.on("config", config => {
-  log("storagePath", config.storagePath);
-  if (config.storagePath === storagePath) return;
-  const prevStoragePath = storagePath;
-  if (server)
-    stopServer(() => {
-      log(`closed server with:
-  storagePath: ${prevStoragePath}`);
-    });
-  storagePath = config.storagePath;
+rnBridge.channel.once("config", config => {
+  log("config", config);
+  if (server) {
+    const error = new Error(
+      "Server already existed when config event was received"
+    );
+    status.setState(constants.ERROR, { error, context: "config" });
+    return;
+  }
   try {
     server = createServer({
       privateStorage: rnBridge.app.datadir(),
-      sharedStorage: storagePath,
       apkPath: config.apkPath,
       apkVersion: config.apkVersion,
+      sharedStorage: config.storagePath,
```

Comment:
```suggestion
```

These are not named the same now, and easier if we only maintain this in createServer() and `src/frontend/api.js` rather than having to keep change this also if the config shape changes